lib/LaTeXML/Post.pm

```diff
@@ -447,7 +447,7 @@ sub convertXMTextContent {
   foreach my $node (@nodes) {
     if ($node->nodeType == XML_TEXT_NODE) {
       my $string = $node->textContent;
-      $string =~ s/^\s+/$NBSP/; $string =~ s/\s+$/$NBSP/;    # should we???
```

Comment:
So, whenever you touch LaTeXML code, you should be asking yourself two questions:
 - Why did Bruce introduce it in LaTeXML?
 - Should I be changing this natively for sTeX?

Have you asked why the non-breaking spaces are there? If not - you should, there may be a good reason. Once we understand the reasoning behind them, then we can think about whether LaTeXML needs them or not, and if it doesn't they will get removed in the LaTeXML master.

We should try to keep KWARC/LaTeXML as close as possible to brucemiller/LaTeXML, the two projects definitely contribute to each other and ideally will one day reintegrate together.